assets/app.js

```diff
@@ -794,39 +788,88 @@ new Vue({
         },
     },
     methods: {
-        validateRejectedProduct: function (type, product) {
-            if (type == 'materials' && product['is_rejected'] && product['rejected_' + type]) {
-                if (!(product['is_rejected'] && product['rejected_' + type])) {
-                    return true
+        validateCacheMaterial: function () {
+            active_languages = this.activeLanguagesIds
+            this.products.forEach(function (product) {
+                active_languages.forEach(function (language) {
+                    materials = product.materials.map(function (material) {
+                        return material[language].trim()
+                    })
+
+                    cached_materials = product.cached_materials[language].value.split('/').map(function (resource) {
+                        return resource.trim();
+                    });
+                    is_overridden = true
+                    if (materials.length === 0) {
+                        is_overridden = false
+                    }
+                    //compare cached_materials and materials
+                    if (cached_materials.length === materials.length) {
+                        is_overridden = false
+                        for (i = 0; i < materials.length; i++) {
+                            if (cached_materials.indexOf(materials[i]) < 0) {
+                                is_overridden = true
+                                break
+                            }
+                        }
+                    }
+
+                    product.cached_materials[language]['is_overridden'] = is_overridden
+                })
+            })
+
+            return this.products
+        },
+        validateRejectedProduct: function (product) {
+            let verified_resources = {
+                'base_product': true,
+                'component1': true,
+                'component2': true,
+                'materials': true
+            }, no_deleted_materials = true
+
+            if (!product['is_rejected']) {
```

Comment:
i don't think we need this check. I think this method is more useful when it also validate not-rejected products